caas/kubernetes/provider/resources/serviceaccount.go

```diff
@@ -90,6 +90,40 @@ func (sa *ServiceAccount) Delete(ctx context.Context, client kubernetes.Interfac
 	return nil
 }
 
+func (sa *ServiceAccount) Ensure(
+	ctx context.Context,
+	client kubernetes.Interface,
+	claims ...Claim,
+) ([]func(), error) {
+	cleanups := []func(){}
+	hasClaim := true
+
+	existing := ServiceAccount{sa.ServiceAccount}
+	err := existing.Get(ctx, client)
+	if err == nil {
+		hasClaim, err = RunClaims(claims...).Assert(&existing.ServiceAccount)
+	}
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
I don't like this error handling here, I would prefer each error to be explicitly handled.

Reply:
Done & Done